lib/streamlit/runtime/fragment.py

```diff
@@ -181,7 +182,10 @@ def wrapped_fragment():
                 # fragment was declared.
                 ctx.cursors = deepcopy(cursors_snapshot)
                 dg_stack.set(deepcopy(dg_stack_snapshot))
+
+                ctx.new_fragment_ids.add(fragment_id)
```

Comment:
@vdonato this fixes the issue where you can see an exception in a full app run when triggering a rerun request 🤔